packages/graphql/src/translate/cypher-builder/build-merge-statement.ts

```diff
@@ -0,0 +1,123 @@
+import { Node, Neo4jGraphQLCypherBuilderError } from "../../classes";
+import { CypherStatement } from "../types";
+import { joinStrings } from "../../utils/utils";
+import { buildNodeStatement } from "./build-node-statement";
+import { joinStatements } from "../utils/join-statements";
+import { buildRelationStatement } from "./build-relation-statement";
+import { generateParameterKey } from "./utils";
+
+type TargetNode = {
+    varName: string;
+    node?: Node;
+    parameters?: Record<string, any>;
+    onCreate?: Record<string, any>;
+};
+
+type TargetRelation = {
+    varName?: string;
+    relationField: RelationField;
+    onCreate?: Record<string, any>;
+};
+
+type MergeNodeArguments = {
+    leftNode: TargetNode & { node: Node };
+    context: Context;
+};
+
+type MergeRelationArguments = {
+    leftNode: TargetNode;
+    rightNode: TargetNode;
+    relation: TargetRelation;
+    context: Context;
+};
+
+export function buildMergeStatement(args: MergeNodeArguments): CypherStatement;
+export function buildMergeStatement(args: MergeRelationArguments): CypherStatement;
+export function buildMergeStatement({
+    leftNode,
+    rightNode,
+    relation,
+    context,
+}: {
+    leftNode: TargetNode;
+    rightNode?: TargetNode;
+    relation?: TargetRelation;
+    context: Context;
+}): CypherStatement {
+    const onCreateStatements: Array<CypherStatement> = [];
+    let leftStatement: CypherStatement | undefined;
+    let relationOnCreateStatement: CypherStatement | undefined;
+
+    if (leftNode.onCreate) {
+        onCreateStatements.push(buildOnCreate(leftNode.onCreate, leftNode.varName));
+    }
+    if (rightNode?.onCreate) {
+        onCreateStatements.push(buildOnCreate(rightNode.onCreate, rightNode.varName));
+    }
+
+    if (relation || rightNode) {
+        if (!relation) {
+            throw new Neo4jGraphQLCypherBuilderError("Missing relation in Cypher merge relation statement");
+        }
+        if (!rightNode) {
+            throw new Neo4jGraphQLCypherBuilderError("Missing rightnode in Cypher merge relation statement");
+        }
+
+        const relationshipName = relation.varName || `${leftNode.varName}_relationship_${rightNode.varName}`;
+        leftStatement = buildRelationStatement({
+            context,
+            leftNode,
+            rightNode,
+            relation: {
+                relationField: relation.relationField,
+                varName: relationshipName,
+            },
+        });
+
+        if (relation.onCreate) {
+            relationOnCreateStatement = buildOnCreate(relation.onCreate, relationshipName);
+            onCreateStatements.push(relationOnCreateStatement);
+        }
+    } else {
+        leftStatement = buildNodeStatement({
+            ...leftNode,
+            context,
+        });
+    }
+
+    const mergeNodeStatement = joinStatements(["MERGE ", leftStatement], "");
+
+    const onCreateSetQuery = onCreateStatements.length > 0 ? "ON CREATE SET" : "";
```

Comment:
Let's add a new line between `ON CREATE` and `SET`,  makes it more consistent with the style established in the official docs: https://neo4j.com/docs/cypher-manual/current/clauses/merge/#merge-merge-with-on-match-setting-multiple-properties